app/Platform/Controllers/BeatenGamesLeaderboardController.php

```diff
@@ -0,0 +1,196 @@
+use Illuminate\Contracts\View\View;
+use Illuminate\Http\Request;
+use Illuminate\Pagination\LengthAwarePaginator;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\DB;
+
+class BeatenGamesLeaderboardController extends Controller
+{
+    private int $pageSize = 25;
+
+    public function __invoke(Request $request): View
+    {
+        if (!config('feature.beat')) {
+            abort(404);
+        }
+
+        $validatedData = $request->validate([
+            'page.number' => 'sometimes|integer|min:1',
+            'filter.system' => 'sometimes|integer',
+            'filter.retail' => 'sometimes|in:true,false',
+            'filter.hacks' => 'sometimes|in:true,false',
+            'filter.homebrew' => 'sometimes|in:true,false',
+            'filter.unlicensed' => 'sometimes|in:true,false',
+            'filter.prototypes' => 'sometimes|in:true,false',
+            'filter.demos' => 'sometimes|in:true,false',
+        ]);
+
+        $targetSystemId = (int) ($validatedData['filter']['system'] ?? 0);
+
+        $gameKindFilterOptions = [
+            'retail' => ($validatedData['filter']['retail'] ?? true) !== 'false',
+            'hacks' => ($validatedData['filter']['hacks'] ?? true) !== 'false',
+            'homebrew' => ($validatedData['filter']['homebrew'] ?? true) !== 'false',
+            'unlicensed' => ($validatedData['filter']['unlicensed'] ?? true) !== 'false',
+            'prototypes' => ($validatedData['filter']['prototypes'] ?? true) !== 'false',
+            'demos' => ($validatedData['filter']['demos'] ?? true) !== 'false',
+        ];
+
+        // Where do I currently rank? This is a separate query that doesn't include the page/offset.
+        $me = Auth::user() ?? null;
+        $myRankingData = null;
+        if ($me) {
+            $myRankingData = $this->buildUserRanking($me, $targetSystemId, $gameKindFilterOptions);
+        }
+
+        // Now get the current page's rows.
+        $currentPage = $validatedData['page']['number'] ?? 1;
+        $offset = (int) ($currentPage - 1) * $this->pageSize;
+        $startingRank = (int) ($currentPage - 1) * $this->pageSize + 1;
+
+        $beatenGameAwardsRankedRows = $this->buildBeatenGameAwardsRankings($offset, $targetSystemId, $gameKindFilterOptions);
+
+        // We also need to know how many rows there are, otherwise the
+        // paginator can't determine what the max page number should be.
+        $rankedRowsCount = $this->buildFilteredGamesQuery($targetSystemId, $gameKindFilterOptions)
+            ->distinct('filteredGames.user')
+            ->count('filteredGames.user');
+
+        $paginator = new LengthAwarePaginator($beatenGameAwardsRankedRows, $rankedRowsCount, $this->pageSize, $currentPage, [
+            'path' => $request->url(),
+            'query' => $request->query(),
+        ]);
+
+        // Grab all the systems so we can build the system filter options.
+        $allSystems = System::orderBy('Name')->get(['ID', 'Name']);
```

Comment:
Events appears to still be a selectable item.
![image](https://github.com/RetroAchievements/RAWeb/assets/32680403/e88900dd-0bf5-4a39-b9d4-18af13132983)

And the list seems to be using something other than `Name` for display. For example, `Name` for console 63 is "Watara Supervision", but it just shows "Supervision" in the dropdown, which seems out of place given the sort was on "Watara".
![image](https://github.com/RetroAchievements/RAWeb/assets/32680403/86001d5f-0d66-4661-9940-063b238d6a93)

Reply:
Using `Name` in latest: https://github.com/RetroAchievements/RAWeb/pull/1862/commits/b113ee264709ee466b4c1087f32adcabd84b3d15#diff-b591e62d226f39ad0c2697521c831ba732612a8dc708272dd6e816a7a22ff7e1R16

Also, "Events" is filtered out now:
https://github.com/RetroAchievements/RAWeb/pull/1862/commits/b113ee264709ee466b4c1087f32adcabd84b3d15#diff-b591e62d226f39ad0c2697521c831ba732612a8dc708272dd6e816a7a22ff7e1R16